django/cantusdb_project/main_app/management/commands/update_cached_concordances.py

```diff
@@ -89,35 +81,79 @@ def get_concordances() -> dict:
         "volpiano",
     )
 
-    stdout.write("Processing chants\n")
-    concordances: defaultdict = defaultdict(list)
-    for chant in values:
-        source_id: int = chant["source_id"]
-        source_absolute_url: str = f"{DOMAIN}/source/{source_id}/"
-        chant_id: int = chant["id"]
-        chant_absolute_url: str = f"{DOMAIN}/chant/{chant_id}/"
-
-        concordances[chant["cantus_id"]].append(
-            {
-                "siglum": chant["source__siglum"],
-                "srclink": source_absolute_url,
-                "chantlink": chant_absolute_url,
-                "folio": chant["folio"],
-                "sequence": chant["c_sequence"],
-                "incipit": chant["incipit"],
-                "feast": chant["feast__name"],
-                "genre": chant["genre__name"],
-                "office": chant["office__name"],
-                "position": chant["position"],
-                "cantus_id": chant["cantus_id"],
-                "image": chant["image_link"],
-                "mode": chant["mode"],
-                "full_text": chant["manuscript_full_text_std_spelling"],
-                "melody": chant["volpiano"],
-                "db": "CD",
-            }
-        )
+    concordances: list[dict] = [make_chant_dict(chant) for chant in values]
+    return concordances
+
+
+def make_chant_dict(chant: dict) -> dict:
+    """Given a dictionary representing a chant from the database,
+    return a chant with the keys specified at
+    https://github.com/DDMAL/CantusDB/wiki/APIs#concordances
+
+    Args:
+        chant (dict): A dictionary representing a chant from the database,
+            from a QuerySet.values() call, with several non-standardized
+            keys from database traversals (e.g., `source__siglum`)
+
+    Returns:
+        dict: A dictionary representing a chant, with several values standardized
+            (e.g., `siglum`) and several values calculated (e.g., `srclink`)
+    """
+    DOMAIN: str = "https://cantusdatabase.org"
```

Comment:
This is a constant and doesn’t need to be redeclared every time the function is called.

Reply:
good call - thank you!